### ISSUE TITLE
Create PropertyChangedEventArgs statically

### ISSUE THREAD

**Issue by ElemarJR** [code-cracker/code-cracker#42]
Implement a C# version of @ljw1004 Analyzer/CodeFix

As he describes it:

> When you implement INotifyPropertyChanged, if one of your properties changes,
> you are expected to fire off a PropertyChanged event. This takes a
> PropertyChangedEventArgs argument. People sometimes construct a New
> instance of the PropertyChangedEventArgs class every single time the
> property changes. This amounts of a lot of heap allocations, which
> hurt performance in XAML apps.
> 
> This analyzer looks for every place where you construct an instance of PropertyChangedEventArgs
> in something other than a static field. Its fix Is to instead construct the instance
> in a static field.
> 
> NOTE: This analyzer doesn't do work to avoid name-clashes. If the static field already
> exists, it will give poor results.

Sample Video in https://pbs.twimg.com/tweet_video/B2tTZRNCIAAHlqx.mp4

Reference implementation (VB) in 

https://github.com/ljw1004/dotnet-code-analyzer/blob/master/dotnet-code-analyzer/Analyzer_XamlAppTips/UseStaticChangedCS.vb

This is essentially a refactoring, so we are classifying it as a `Hidden` diagnostic.
Diagnostic Id: `CC0106`

This:

``` csharp
private string name;
public string Name
{
    get
    {
        return name;
    }

    set
    {
        name = value;
        PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(nameof(Name));
    }
}
```

Becomes:

``` csharp
private string name;
public string Name
{
    get
    {
        return name;
    }

    set
    {
        name = value;
        PropertyChanged?.Invoke(this, nameChanged);
    }
}
private static PropertyChangedEventArgs nameChanged = new PropertyChangedEventArgs(nameof(Name);
```

If there is already a field named `nameChanged`, add `1` to it, if there is already on with `1`, add `2`, and so on.

Diagnostic Id; `CC0106`

**Comment by giggio** [comment]
I added the test case and diagnostic id. It is up for grabs.

**Comment by baks** [comment]
@giggio, this one looks interesting. I would like to take care of analyzer/code-fix.

**Comment by giggio** [comment]
Go for it.

**Comment by giggio** [comment]
Closed on #760.
